rust-runtime/aws-smithy-http-server/src/rejection.rs

```diff
@@ -3,231 +3,222 @@
+//! modified at any time without causing breaking changes. They are not surfaced to clients or the
+//! service implementer in any way (including this documentation): indeed, they can't be converted
+//! into responses. They serve as a mechanism to keep track of all the possible errors that can
+//! occur when processing a request or a response, in far more detail than what AWS protocols need
+//! to. This is why they are so granular: other (possibly protocol-specific) error types (like
+//! [`crate::runtime_error::RuntimeError`]) can "group" them when exposing errors to
+//! clients while the framework does not need to sacrifice fidelity in private error handling
+//! routines, and future-proofing itself at the same time (for example, we might want to record
+//! metrics about rejection types).
+//!
+//! Rejection types implement [`std::error::Error`], and some take in type-erased boxed errors
+//! (`crate::Error`) to represent their underlying causes, so they can be composed with other types
+//! that take in (possibly type-erased) [`std::error::Error`]s, like
+//! [`crate::runtime_error::RuntimeError`], thus allowing us to represent the full
+//! error chain.
+
+use strum_macros::Display;
+
+/// Rejection used for when failing to extract an [`crate::Extension`] from an incoming [request's
+/// extensions]. Contains one variant for each way the extractor can fail.
+///
+/// [request's extensions]: https://docs.rs/http/latest/http/struct.Extensions.html
+#[derive(Debug, Display)]
+pub enum RequestExtensionNotFoundRejection {
+    /// Used when a particular [`crate::Extension`] was expected to be found in the request but we
+    /// did not find it.
+    /// This most likely means the service implementer simply forgot to add a [`tower::Layer`] that
+    /// registers the particular extension in their service to incoming requests.
+    MissingExtension(String),
+    // Used when the request extensions have already been taken by another extractor.
+    ExtensionsAlreadyExtracted,
+}
+
+impl std::error::Error for RequestExtensionNotFoundRejection {}
+
+/// Errors that can occur when serializing the operation output provided by the service implementer
+/// into an HTTP response.
+#[derive(Debug, Display)]
+pub enum ResponseRejection {
+    /// Used when `httpResponseCode` targets an optional member, and the service implementer sets
+    /// it to `None`.
+    MissingHttpStatusCode,
+
+    /// Used when the service implementer provides an integer outside the 100-999 range for a
+    /// member targeted by `httpResponseCode`.
+    InvalidHttpStatusCode,
+
+    /// Used when an invalid HTTP header value (a value that cannot be parsed as an
+    /// `[http::header::HeaderValue]`) is provided for a shape member bound to an HTTP header with
+    /// `httpHeader` or `httpPrefixHeaders`.
+    /// Used when failing to serialize an `httpPayload`-bound struct into an HTTP response body.
+    Build(crate::Error),
```

Comment:
Are you sure about this documentation? The first 3 lines look to be for something else.. The name `Build` is also not very significative to me.

Reply:
Ah ok, got it. Thanks for the explanation.